mypy/typestate.py

```diff
@@ -0,0 +1,55 @@
+"""
+A shared state for all TypeInfo's that holds global cache and dependency information,
+and potentially other mutable TypeInfo state.
+"""
+
+from typing import Dict, Set, Tuple
+
+MYPY = False
+if MYPY:
+    from typing import ClassVar
+    from mypy.nodes import TypeInfo
+    from mypy.types import Instance
+
+
+class TypeState:
```

Comment:
Add docstring. In the docstring, maybe discuss at least these things:

* Purpose of this class.
* How to avoid leaking global state: this needs to be explicitly cleared after a build has finished and after a daemon shutdown.
* Invariants: Resetting subtype caches for a class has no semantic effect (only perf).